tests/mssqltestutils.py

```diff
@@ -90,29 +91,62 @@ def getTempPath(*args):
     return  os.path.abspath(tempPath)
 
 def create_test_db():
+    """
+    Creates database for test, using various status checks and retry logic for reliability.
+    - Calls helper method to check status of create db task, if possible
+    - Exits on successful response or retry period exceeds time limit
+    """
     client = create_mssql_cli_client()
     local_machine_name = socket.gethostname().replace(
         '-', '_').replace('.', '_')
+
     test_db_name = u'mssqlcli_testdb_{0}_{1}'.format(
         local_machine_name, random_str())
     query_db_create = u"CREATE DATABASE {0};".format(test_db_name)
-    count = 0
-
-    # retry logic in case db create fails
-    while count < 5:
-        for _, _, status, _, is_create_error in client.execute_query(query_db_create):
-            if not is_create_error:
-                shutdown(client)
-                return test_db_name
-            # log warning to console
+
+    for _, _, status, _, is_create_error in client.execute_query(query_db_create):
+        create_db_status = check_create_db_status(test_db_name, client)
+
+        if is_create_error or create_db_status == 'FAILED':
+            # log warning to console and cleanup db
             warnings.warn('Test DB create failed with error: {0}'.format(status))
-        count += 1
+            clean_up_test_db(test_db_name)
+
+        elif create_db_status in ('COMPLETED', 'UNKNOWN'):
+            # 'unknown' is returned if the check_create_db_status function creates db
+            # using a non-master db. only master db is supported.
+            shutdown(client)
+            return test_db_name
+
     shutdown(client)
 
     # cleanup db just in case, then raise exception
     clean_up_test_db(test_db_name)
     raise AssertionError("DB creation failed.")
 
+def check_create_db_status(db_name, client):
+    """
+    Uses retry logic with sys.dm_operation_status to check statis of create database job.
+    """
+    if client.database.lower() != 'master':
+        # sys.dm_operation_status CANNOT be used if the client isn't running on master db.
+        return 'UNKNOWN'
```

Comment:
Probably should throw and error here as it's probably a coding mistake.

Reply:
Changed functionality here to raise an exception. Create db method now catches the raised exception instead of checking for `'UNKNOWN'`.